astropy/cosmology/core.py

```diff
@@ -146,6 +146,9 @@ def _register_cls(cls):
         # register class
         _COSMOLOGY_CLASSES[cls.__qualname__] = cls
 
+        if inspect.isabstract(cls):  # skip abstract classes
```

Comment:
Should it be added to `_COSMOLOGY_CLASSES`, though? Also, which are the abstract classes that are skipped? I'm surprised this helps! Maybe expand the comment...

Reply:
> Should it be added to _COSMOLOGY_CLASSES, though?

They were before, so I think yes

> Also, which are the abstract classes that are skipped?

`Cosmology` and `FLRW`

> I'm surprised this helps! 

It didn't 😆. But I found the real culprit.

Reply:
Ha! Maybe still add an explicit comment here that it is to skip registering `Cosmology, FLRW`.

Reply:
Is this patch still necessary given the updated check in test module above?

Reply:
Yes. Abstract classes cannot be initialized, so shouldn't be registered.

Reply:
That also suggests they should not be in `_COSMOLOGY_CLASSES`. But since they were before, that probably doesn't hurt. Still, do it earlier? If so, I'd suggest moving the `if` statement inside `__init_subclass__`.